src/filters/FilterSystem.ts

```diff
@@ -243,15 +250,21 @@ export class FilterSystem implements System
             return;
         }
 
-        const viewPort = renderer.renderTarget.rootViewPort;
-
         // here we constrain the bounds to the viewport we will render too
         // this should not take into account the x, y offset of the viewport - as this is
         // handled by the viewport on the gpu.
         // need to factor in resolutions also..
         bounds
-            .scale(resolution)
-            .fitBounds(0, viewPort.width, 0, viewPort.height)
+            .scale(resolution);
+
+        if (clipToViewport)
+        {
+            const viewPort = renderer.renderTarget.rootViewPort;
+
+            bounds.fitBounds(0, viewPort.width, 0, viewPort.height);
+        }
+
+        bounds
```

Comment:
Not sure why do we need to scale it, ceil and then scale it back..
Do we need it even if we do not clip bounds into viewport?

Reply:
good question! The scale is to deal with precision issues when rendering to different resolutions - basically we need to round them correctly otherwise you get a jittering effect.